homeassistant/components/device_tracker/google_maps.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for Google Maps location sharing.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.google_maps/
+"""
+import logging
+import os
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, SOURCE_TYPE_GPS)
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['locationsharinglib==0.2.1']
+
+CONF_IGNORED_DEVICES = 'ignored_devices'
+CONF_ACCOUNTNAME = 'account_name'
+
+# CREDENTIALS_FILE = 'google_maps_location_sharing.conf'
+CREDENTIALS_FILE = 'cookies.pickle'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_scanner(hass, config: dict, see, discovery_info=None):
```

Comment:
There's a special `ConfigType` in [helpers/typing](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/typing.py) that should probably be used instead of `dict`.